front/app/components/admin/ProjectableHeaderBgUploader/index.tsx

```diff
@@ -0,0 +1,76 @@
+import React, { useEffect, useState } from 'react';
+
+import { Box } from '@citizenlab/cl2-component-library';
+import { UploadFile } from 'typings';
+import ImagesDropzone from 'components/UI/ImagesDropzone';
+
+import { convertUrlToUploadFile } from 'utils/fileUtils';
+import { HEADER_BG_ASPECT_RATIO } from 'services/projects';
+import ImageCropperContainer from 'components/admin/ImageCropper/Container';
+import ImageInfoTooltip from 'components/admin/ImageCropper/ImageInfoTooltip';
+import { SectionField, SubSectionTitle } from 'components/admin/Section';
+import { FormattedMessage } from 'utils/cl-intl';
+import messages from './messages';
+
+interface Props {
+  imageUrl: string | null | undefined;
+  onImageChange: (newImageBase64: string | null) => void;
+}
+
+const HeaderBgInput = ({ imageUrl, onImageChange }: Props) => {
+  const [headerBg, setHeaderBg] = useState<UploadFile[] | null>(null);
+  useEffect(() => {
+    (async () => {
+      if (imageUrl) {
+        const headerFile = await convertUrlToUploadFile(imageUrl, null, null);
+        setHeaderBg(headerFile ? [headerFile] : null);
+      }
+    })();
+  }, [imageUrl]);
+
+  const handleImageAdd = (newHeader: UploadFile[]) => {
+    const newHeaderFile = newHeader[0];
+
+    onImageChange(newHeaderFile.base64);
+    setHeaderBg([newHeaderFile]);
+  };
+
+  const handleImageRemove = async () => {
+    onImageChange(null);
+    setHeaderBg(null);
+  };
+
+  const imageIsNotSaved = headerBg && !headerBg[0].remote;
```

Comment:
Having "not" in a variable name is often harder to understand. Easier is to do imageIsSaved and use !imageIsSaved if you want to test it not being saved.

Reply:
I'm not sure `!imageIsSaved` is easier to read than `imageIsNotSaved`. I could switch the cropper and the dropzone, but the condition would look like that
```ts
const imageIsSaved = !(headerBg && !headerBg[0].remote);
```
And it doesn't seem easier to read.

Reply:
Or how about this?

>  const imageIsSaved = headerBg ? headerBg[0].remote : false;

Reply:
Btw, I think we can also stop putting these single images into arrays. I'm also fixing that in an upcoming PR.

Reply:
No, I was confused, `const imageIsSaved = headerBg ? headerBg[0].remote : false;` is not the same as before. `imageIsNotSaved` is false when `headerBg` is null. Then `imageIsSaved` should be true in this case.

In general, the variable name seems to be misleading. I've renamed it to `imageShouldBeSaved` 5e4be0af6f

Reply:
In my version, when headerBg is null, imageIsSaved is also false. It should not be true I believe. Because if there is no image, it has not been saved.

I think it's correct what we had. If there is an image, we check if its remote prop is true so we now it's saved, otherwise it's a local image (not saved yet). Otherwise, with no image (it being null), we don't have a saved image as there is none.

Reply:
Just looked at your code: I also like the name you picked, it's clear to me 👍 The only problem, type-wise, is that sometimes it's type will be an image, which is kind of unexpected. We want it to be a boolean, which is why I prefer a ternary operator.

Reply:
(just removed my first confusing comment)